cluster-autoscaler/cloudprovider/huaweicloud/huaweicloud_service_manager.go

```diff
@@ -86,6 +92,7 @@ type cloudServiceManager struct {
 	cloudConfig      *CloudConfig
 	getECSClientFunc func() *huaweicloudsdkecs.EcsClient
 	getASClientFunc  func() *huaweicloudsdkas.AsClient
+	asgs             *autoScalingGroups
```

Comment:
I'm not sure if the cloud service manager should hold a cache here.

A scaling group already holds a service manager, and the service manager also holds a scaling group cache, the cache holds all scaling groups. Here is a pointer loop, don't you think?

Reply:
I don't really think it is a pointer loop. A scaling group doesn't hold a service manager, scaling groups and service manager are blong to provider.

Why I created another scaling group under manager mainly because scaling groups hold by provider were created by specs, which should be an attribute of provider, while scaling groups hold by service manager is actually used for caching instance, and those groups were reconstructed by us and have specific usages, so it should be hold by manager I think.

Reply:
OK.